### ISSUE TITLE
About DA3-Long

### ISSUE THREAD

**Issue by yulongg1** [ByteDance-Seed/Depth-Anything-3#132]
Hi, thanks for your great work on Depth Anything 3!

I have two questions:

1. When will the code for DA3-Long be released?

I saw the mention of the “long” version in the paper and demos, and I’m wondering if there is an expected timeline for releasing the code or pretrained models for DA3-Long.

2. How to obtain metric-scaled point clouds when using DA3-Long without the pose-conditioned mode?

If I don’t use the pose-conditioned mode, but I do have prior poses or trajectories that already contain metric scale (e.g., from GPS/INS/visual odometry), is there a way to enforce or recover metric scale in the depth predictions produced by DA3-Long?

In other words, is there a recommended way to combine DA3-Long with externally scaled poses so that the resulting point clouds also maintain real-world scale?

Thanks!

**Comment by Jaykumaran** [comment]
+1 

Same here, eagerly expecting DA3-Long release and to make use of it for large scale 3DGS / SLAM training

**Comment by jckhng** [comment]
I've just modified vggt_long.py to use DA3's api. You can try it here:
https://github.com/jckhng/VGGT-Long

So just install DA3 and run `python da3_long.py --image_dir path/to/images`

The relevant code is here: https://github.com/jckhng/VGGT-Long/blob/03f6fcb47abc5a761b1dd5dc5de4ae2f5764e358/da3_long.py

**Comment by Jaykumaran** [comment]
Hi @jckhng , thanks for sharing this with us. From your experiments, were able to achieve on par or better results than vggt_long or pi_long v/s da3_long. Could you please share some artifacts like trajectory plot (loop closure) and the combined ply file o/p visulization.

Kindly share your insights and some results comparison on the same.  (if benchmarks available)

Also is the colmap export of the same working for 3DGS training  in your case?!

Thank you so much.

**Comment by Jaykumaran** [comment]
Hey, i tried with my dataset of where i walk linearly along a road, i think the poses werent correct when i used your DA3 wrapper, however for same sequence vggt long results were all good, I'. using the default configs provided by base yaml





VGGT Long: (correct trajectory )
<img width="2074" height="1638" alt="Image" src="https://github.com/user-attachments/assets/e1c4d270-4e2c-442c-b738-85718d16cb9e" />

<img width="1200" height="631" alt="Image" src="https://github.com/user-attachments/assets/b0f45fc9-6006-4db2-afc9-feef1242aaa3" />




DA3 Long Trajectory: 


<img width="2060" height="1638" alt="Image" src="https://github.com/user-attachments/assets/c9f6e3e3-252f-4913-89df-8ceea2ff344c" />

<img width="978" height="897" alt="Image" src="https://github.com/user-attachments/assets/4888370c-d94a-4bab-800f-d13757481257" />


```
Weights:
  VGGT: './weights/model.pt'
  DBoW: './weights/ORBvoc.txt'
  SALAD: './weights/dino_salad.ckpt'
  DNIO: './weights/dinov2_vitb14_pretrain.pth'

Model:
  chunk_size: 60
  overlap: 30
  loop_chunk_size: 20 # imgs of loop chunk = 2 * loop_chunk_size
  loop_enable: True
  useDBoW: False
  delete_temp_files: True
  align_lib: 'numba' # 'numba' or 'numpy', use 'numba' would be faster
  align_method: 'sim3' # choose among 'sim3', 'se3' or 'scale+se3'
  # find bugs related to the 'scale+se3'. we are fixing them, will get done in days. [by Deng]
  scale_compute_method: 'auto' # choose among 'auto', 'ransac' or 'weighted'. It is used when align_method == 'scale+se3'
  align_type: 'dense' # choose between 'dense' or 'sparse'. Sparse align is constructing, not avaible for now
  
  reference_frame_mid: False 
  # If reference_frame_mid == True, use the middle frame within the chunk as the 
  # reference frame to alleviate the scale drift when loop closure is not available.
  # Disable it for now cuz i found some bugs related to this feature [by deng on 09 Oct].
  
  Sparse_Align:
    keypoint_select: 'orb' # choose among 'orb', 'fast' or 'disk'
    keypoint_num: 5000

  IRLS:
    delta: 0.1
    max_iters: 5
    tol: 1e-9

  Pointcloud_Save:
    sample_ratio: 0.015
    conf_threshold_coef: 0.75 # conf_threshold = np.mean(confs) * conf_threshold_coef

Loop:
  DBoW:
    thresh: 0.034
    num_repeat: 3
    NMS: 25

  SALAD:
    image_size: [336, 336]
    batch_size: 32
    similarity_threshold: 0.7
    top_k: 5
    use_nms: True
    nms_threshold: 25

  SIM3_Optimizer:
    lang_version: 'cpp' 
    # choose between 'cpp' or 'python'. will auto set 'python' if c++ version has not installed
    max_iterations: 30
    lambda_init: 1e-6
```

**Comment by jckhng** [comment]
Hi, from the looks of it, there is some error introduced by me during loop optimation. I will look into it. Meantime you can disable loop optimization.

loop_enable: False

I'm midst of testing if I can use the colmap's cameras.txt and images.txt for 3DGS.

**Comment by Jaykumaran** [comment]
Hello again, 

The actual movement was a 20m walk forward --> turn back --> same lane walk back ~15m

By enabling loop : False

1. Doubling effect  -- which is understood given loop closure disabled 
<img width="1634" height="984" alt="Image" src="https://github.com/user-attachments/assets/9ae0232e-3966-4004-ac61-ff54b6987613" />

2. Colors doesnt look right, -- looks negative -- inverted colors,, checked even inidividual chunks, same issue

<img width="1137" height="795" alt="Image" src="https://github.com/user-attachments/assets/7ae3ad37-0166-44b3-8646-07ce03ae003c" />

**Comment by jckhng** [comment]
I have just updated the code to fix the colors. loop_enable is still broken. 

DA3 does not provide world_points and world_points_conf, so the VGGT-Long's approach of aligning based on high confidence points may not be the best suited here.

**Comment by Jaykumaran** [comment]
Understood, kindly ping and share here if you are able to find a fix.  Tqsm
